### ISSUE TITLE
module.rule.generator.publicPath not work for css background-image

### ISSUE THREAD

**Issue by JasonBoy** [web-infra-dev/rspack#4956]
### System Info

System:
    OS: macOS 13.6
    CPU: (8) x64 Intel(R) Core(TM) i7-8557U CPU @ 1.70GHz
    Memory: 29.29 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
    pnpm: 8.7.1 - /usr/local/bin/pnpm
  Browsers:
    Chrome: 119.0.6045.199
    Chrome Canary: 122.0.6171.0
    Safari: 16.6

### Details

when use `publicPath` in rule, the images from css/less/vue-style tag or the `style` in template still prefixed with the `output.publicPath`, not the inner one；but in `img` tag with `src`, it works:
```js
module.exports = {
  //...
  output: {
    publicPath: 'https://xxx.com/',
  },
  module: {
    rules: [
      {
        test: /\.png/,
        type: 'asset/resource',
        generator: {
          publicPath: 'https://cdn.xxx.com/',
        },
      },
    ],
  },
};
```

### Reproduce link

[rspack-vue-publicPath-reproduce](https://github.com/JasonBoy/rspack-vue-publicpath-reproduce)

### Reproduce Steps

in my case it's a vue2 project with less:

write a tag like`<p style="background-image: url('@/assets/tips.png')">name</p>` or
in the less file or vue/style tag:
```css
p {
  background-image: url('@/assets/tips.png')
}
```
cases above still use the `output.publicPath`

**Comment by JasonBoy** [comment]
<img width="1341" alt="image" src="https://github.com/web-infra-dev/rspack/assets/2911620/b5b8361a-8106-42bf-9d67-7024a7e2445d">

**Comment by faga295** [comment]
It seems to be because the `loaderContext` is missing the `importModule`, maybe https://github.com/web-infra-dev/rspack/pull/4718 this pr will support it

**Comment by JasonBoy** [comment]
> It seems to be because the `loaderContext` is missing the `importModule`, maybe this pr(#4718) will support it

@faga295  any EST for this pr to be merged and released? I would like to have a try

**Comment by faga295** [comment]
@JSerFeng cc

**Comment by ahabhgk** [comment]
<img width="1084" alt="Screenshot 2023-12-21 at 19 33 39" src="https://github.com/web-infra-dev/rspack/assets/42857895/32b43c33-37e3-4dbf-83f1-c780c3307600">
Look like the img src is correct, I can't reproduce the bug, can you provide more information?

**Comment by JasonBoy** [comment]
@ahabhgk the img src is ok, but the url in `background-image: url("")` is not replaced

**Comment by faga295** [comment]
https://github.com/webpack-contrib/mini-css-extract-plugin/blob/7ddf35a178a6a0dc331e8099e29437f20d0a2c47/src/loader.js#L319-L339 In mini-css-extract-plugin, it will resolve request via importModule, but current rspack is missing it

**Comment by ahabhgk** [comment]
This is a bug about experiments.css url, this will be fixed by #5099, and the fix will land in v0.4.5 in next week

**Comment by JasonBoy** [comment]
> This is a bug about experiments.css url, this will be fixed by #5099, and the fix will land in v0.4.5 in next week

looking forward to it 👀
